datalad/distribution/get.py

```diff
@@ -0,0 +1,224 @@
+            constraints=EnsureStr() | EnsureNone()),
+        source=Parameter(
+            args=("-s", "--source",),
+            metavar="LABEL",
+            doc="""label of the data source to be used to fulfill the request.
+            This can be the name of a dataset :term:`sibling` or another known
+            source""",
+            constraints=EnsureStr() | EnsureNone()),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+        git_opts=git_opts,
+        annex_opts=annex_opts,
+        annex_get_opts=annex_get_opts)
+
+    @staticmethod
+    @datasetmethod(name='get')
+    def __call__(
+            path=None,
+            source=None,
+            dataset=None,
+            recursive=False,
+            recursion_limit=None,
+            git_opts=None,
+            annex_opts=None,
+            annex_get_opts=None):
+
+        # resolve dataset:
+        ds = require_dataset(dataset, check_installed=True,
+                             purpose='getting content')
+        lgr.debug("Resolved dataset: %s" % ds)
+
+        # check parameters:
+        if path is None:
+            raise InsufficientArgumentsError("insufficient information for "
+                                             "getting: requires at least a "
+                                             "path.")
+        # When called from cmdline `path` will be a list even if
+        # there is only one item.
+        # Make sure we deal with the same when called via python API:
+        if not isinstance(path, list):
+            path = [path]
+
+        # resolve path(s):
+        lgr.info("Resolving paths ...")
+        resolved_paths = [resolve_path(p, dataset) for p in path]
+
+        # resolve associated datasets:
+        lgr.info("Resolving (sub-)datasets ...")
+        resolved_datasets = dict()
+        for p in resolved_paths:
+            if not lexists(p):
+                # skip early:
+                # Note/TODO: This is to be changed, when implementing implicit
+                # install of subdatasets
+                lgr.warning("{0} not found. Ignored.".format(p))
+                continue
+
+            p_ds = ds.get_containing_subdataset(p,
+                                                recursion_limit=recursion_limit)
+            if p_ds is None:
+                lgr.warning("{0} not in dataset. Ignored.".format(p))
+                continue
+            if not recursive and p_ds != ds:
+                lgr.warning("{0} belongs to subdataset {1}. To get its content "
+                            "use option `recursive` or call get on the "
+                            "subdataset. Ignored.".format(p, p_ds))
+                continue
+
+            resolved_datasets[p_ds.path] = \
+                resolved_datasets.get(p_ds.path, []) + [p]
+
+            # TODO: Change behaviour of Dataset: Make subdatasets singletons to
+            # always get the same object referencing a certain subdataset.
```

Comment:
I would worry about that later whenever we actually run into the problem... for now I think the most important is to get `get` and `install` working correctly